src/Compilers/CSharp/Test/Semantic/Semantics/ConstantTests.cs

```diff
@@ -3509,37 +3509,176 @@ void M()
-            );
+                Diagnostic(ErrorCode.ERR_NotConstantExpression, @"$""{I1} the {S1}""").WithArguments("F2").WithLocation(12, 27));
+        }
+
+        [Fact]
+        public void ConstantInterpolatedStringsHybrid()
+        {
+            string source = @"
+class C
+{
+    void M()
+    {
+        const string S1 = $""Number "" + ""3"";
+        const string S2 = $""{""Level 5""} "" + S1;
+        const string F1 = $""{S1}"";
+    }
+}";
+            var actual = ParseAndGetConstantFoldingSteps(source);
+
+            var expected =
+@"$""Number "" + ""3"" --> Number 3
+$""Number "" --> Number 
+$""{""Level 5""} "" + S1 --> Level 5 Number 3
+$""{""Level 5""} "" --> Level 5 
+$""{S1}"" --> Number 3";
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public void ConstantInterpolatedStringsHybridError()
+        {
+            string source = @"
+class C
+{
+    void M()
+    {
+        
+        string NC1 = ""Teleporter"";
+        const string S1 = ""The"" + $""Number {3}"" + ""Level 5"";
+        const string S2 = $""Level 4 "" + NC1;
+        const string F1 = $""{S1}"";
+    }
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics(
+                // (8,27): error CS0133: The expression being assigned to 'S1' must be constant
+                //         const string S1 = "The" + $"Number {3}" + "Level 5";
+                Diagnostic(ErrorCode.ERR_NotConstantExpression, @"""The"" + $""Number {3}"" + ""Level 5""").WithArguments("S1").WithLocation(8, 27),
+                // (9,27): error CS0133: The expression being assigned to 'S2' must be constant
+                //         const string S2 = $"Level 4 " + NC1;
+                Diagnostic(ErrorCode.ERR_NotConstantExpression, @"$""Level 4 "" + NC1").WithArguments("S2").WithLocation(9, 27),
+                // (10,27): error CS0133: The expression being assigned to 'F1' must be constant
+                //         const string F1 = $"{S1}";
+                Diagnostic(ErrorCode.ERR_NotConstantExpression, @"$""{S1}""").WithArguments("F1").WithLocation(10, 27));
+        }
+
+        [Fact]
+        public void ConstantInterpolatedStringsVersionError()
+        {
+            string source = @"
+class C
+{
+    #pragma warning disable CS0219 // unused locals
+    void M1()
+    {
+        const string S1 = $""Testing"";
+        const string S2 = $""{""Level 5""} {""Number 3""}"";
+        const string S3 = $""{$""{""Spinning Top""}""}"";
+        const string S4 = $""Hybrid"" + ""Testing"" + ""123"";
+        const string S5 = ""Hybrid"" + ""Testing"" + $""321"";
+        const string F1 = $""{S1}"";
+        const string F2 = F1 + $"" the {S2}"";
+
+        string VS = ""Change"";
+        const string S6 = $""Failed to {VS}"";
+    }
+
+    void M2(string S1 = $""Testing"")
```

Comment:
Why is this test succeeding? I would expect to see failures in version 8 for both this location and the one below.

If you want to just mark them with a prototype comment and address in a follow-up, that's fine. But there needs to be an error :)

Reply:
Huh, you're absolutely right. Those shouldn't be as they are. I suspect there's a separate path that checks for constants in those cases, or maybe it was somewhere higher up the chain. I'll take a look. If I can't find it by noon, I'll slap a PROTOTYPE on it.